mobile/src/stateMachine/views/ViewAffirmation.tsx

```diff
@@ -1,71 +1,105 @@
 import { ViewState } from './base';
 import React from 'react';
-import { NotificationRequest } from 'expo-notifications';
 import { observer } from 'mobx-react';
-import { StyleSheet, Text } from 'react-native';
+import { StyleSheet, Text, Image } from 'react-native';
 import { MasloPage, Container, Button } from 'src/components';
 import { ScenarioTriggers } from '../abstractions';
 import AppController from 'src/controllers';
 import { observable } from 'mobx';
+import { Affirmation } from 'src/constants/QoL';
 
+const affirmationSources = [
+  require('../../assets/gif/affirmation1.gif'),
+  require('../../assets/gif/affirmation2.gif'),
+  require('../../assets/gif/affirmation3.gif'),
+  require('../../assets/gif/affirmation4.gif'),
+  require('../../assets/gif/affirmation5.gif'),
+  require('../../assets/gif/affirmation6.gif'),
+];
 @observer
 export class ViewAffirmationView extends ViewState {
 
   private get notificationsController() {
     return AppController.Instance.User.notifications;
   }
   @observable
-  private currentNotification: NotificationRequest = this.notificationsController.openedNotification;
+  public currentAffirmation: Affirmation;
 
   constructor(props) {
     super(props);
     this._contentHeight = this.persona.setupContainerHeightForceScroll();
   }
 
   async start() {
-    this.currentNotification = this.notificationsController.openedNotification;
+    // When opening this screen from a tap on the notification, this method is called before the Firebase network connection has been enabled, we add a loop and delay here to wait
+    for (let i = 0; i < 200; i++) {
```

Comment:
@MarkDavies0 Why is the loop and sleep needed here? Wouldn't await be sufficient? And wouldn't the loop cause `this.currentAffirmation` to keep getting overwritten before the promise has had a chance to get fulfilled (in which case it's guaranteed to loop 200 times - you might as well sleep for 8000ms from the beginning before calling the awaiting promise to reduce the network calls from 200 to 1).

Reply:
@wynnset The issue is the network call fails because this method is called before the App-wide Firebase network connection has been enabled. I see what you mean about `this.currentAffirmation` getting overwritten. This code is reworked in #174 with the move to userState `getAffirmationToShow` is no longer async

Reply:
Hey @MarkDavies0 I checked #174 and [this piece of code is still there](https://github.com/bipolarbridges/companion-app/blob/48357f52855923918647cfc1ff737a97836a12ee/mobile/src/stateMachine/views/ViewAffirmation.tsx#L33-L47). I thought you said it's reworked in that PR?

Reply:
@wynnset I am currently reworking it to use the userState `synced` event handler to avoid the for loop with sleep. Having some difficulties so I haven't pushed my changes on 174 yet